.github/workflows/tests_and_cov.yml

```diff
@@ -14,21 +14,19 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python 3.12
+      - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.12
+          python-version-file: ".python-version"
 
-        # Cache package installation step to speed up the following step
-      - uses: actions/cache@v4
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          path: ${{ env.pythonLocation }}
-          key: ${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements-dev.txt') }}
+          enable-cache: true
+          cache-dependency-glob: "uv.lock"
```

Comment:
Use cache from the uv action, and only update if `uv.lock` changes. `uv` is so fast anyway it doesn't matter much, but still.